tools/traffic_vault_migrate/traffic_vault_migrate.go

```diff
@@ -0,0 +1,453 @@
+	}
+	sort.Slice(sslkeys[:], func(a, b int) bool {
+		return sslkeys[a].CDN < sslkeys[b].CDN && sslkeys[a].DeliveryService < sslkeys[b].DeliveryService && sslkeys[a].Version < sslkeys[b].Version
+	})
+	sort.Slice(dnssec[:], func(a, b int) bool {
+		return dnssec[a].CDN < dnssec[b].CDN
+	})
+	sort.Slice(uri[:], func(a, b int) bool {
+		return uri[a].DeliveryService < uri[b].DeliveryService
+	})
+	sort.Slice(url[:], func(a, b int) bool {
+		return url[a].DeliveryService < url[b].DeliveryService
+	})
+	return sslkeys, dnssec, uri, url, nil
+}
+
+// UnmarshalConfig takes in a config file and a type and will read the config file into the reflected type
+func UnmarshalConfig(configFile string, t reflect.Type) (reflect.Value, error) {
+	data, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return reflect.Value{}, err
+	}
+	val := reflect.New(t)
+	err = json.Unmarshal(data, val.Interface())
+	if err != nil {
+		return reflect.Value{}, err
+	}
+
+	return val, nil
+}
+
+// TVBackend represents a TV backend that can be have data migrated to/from
+type TVBackend interface {
+	// Start initiates the connection to the backend DB
+	Start() error
+	// Stop terminates the connection to the backend DB
+	Stop() error
+	// Ping checks the connection to the backend DB
+	Ping() error
+	// ValidateKey validates that the keys are valid (in most cases, certain fields are not null)
+	ValidateKey() []string
+	// Name returns the name for this backend
+	Name() string
+	// ReadConfig takes in a filename and will read it into the backends config
+	ReadConfig(string) error
+	// String returns a high level overview of the backend and its keys
+	String() string
+
+	// Fetch gets all of the keys from the backend DB
+	Fetch() error
+	// Insert takes the current keys and inserts them into the backend DB
+	Insert() error
+
+	// GetSSLKeys converts the backends internal key representation into the common representation (SSLKey)
+	GetSSLKeys() ([]SSLKey, error)
+	// SetSSLKeys takes in keys and converts & encrypts the data into the backends internal format
+	SetSSLKeys([]SSLKey) error
+
+	// GetDNSSecKeys converts the backends internal key representation into the common representation (DNSSecKey)
+	GetDNSSecKeys() ([]DNSSecKey, error)
+	// SetDNSSecKeys takes in keys and converts & encrypts the data into the backends internal format
+	SetDNSSecKeys([]DNSSecKey) error
+
+	// GetURISignKeys converts the pg internal key representation into the common representation (URISignKey)
+	GetURISignKeys() ([]URISignKey, error)
+	// SetURISignKeys takes in keys and converts & encrypts the data into the backends internal format
+	SetURISignKeys([]URISignKey) error
+
+	// GetURLSigKeys converts the backends internal key representation into the common representation (URLSigKey)
+	GetURLSigKeys() ([]URLSigKey, error)
+	// SetURLSigKeys takes in keys and converts & encrypts the data into the backends internal format
+	SetURLSigKeys([]URLSigKey) error
+}
+
+type CommonRecord struct{}
```

Comment:
What's the point of this type?

Reply:
Used to hold Last Updated, forgot to remove it.